crates/ruff_cache/src/globset.rs

```diff
@@ -0,0 +1,14 @@
+use crate::{CacheKey, CacheKeyHasher};
+use globset::{Glob, GlobMatcher};
+
+impl CacheKey for GlobMatcher {
```

Comment:
I guess in some sense it could make sense for these implementations to go in (e.g.) `ruff`, or whichever crate actually needs to cache these types, but I think we can't do that due to the orphan rule?

Reply:
Yeah, orphan rule strikes again ;) 

I thought about adding feature flags, but it felt overkill to add it for so little code. 

This is probably the main downside of this approach. `Hash` is implemented by many libraries. Implementing `CacheKey` for non-std types requires adding all of them to `ruff_cache` OR introducing new type wrappers.

Reply:
Yeah. We could add a flag for each "external library", right? Like this crate could have a `globset` flag?

I think it's fine. For the reasons you described, it's actually a good thing that we have to intentionally implement `CacheKey`. We should be wary of putting complex types in settings anyway, since they have to be user-editable, serializable, etc.